adr/adr-001-checkpoint-snarks.md

```diff
@@ -0,0 +1,109 @@
+2. Potential performance overhead in generating recursive SNARKs, especially in Stage 1.
+
+### Risks
+
+1. Recursive SNARK generation might be computationally intensive and time-consuming, especially in Stage 1.
+2. Potential vulnerabilities in the recursive SNARK implementation could compromise the entire history (also a ).
+3. Compatibility issues between different SNARK systems used for epochs and checkpoints (Transition to Stage 2 - if not well thought through - may introduce compatibility issues with existing proofs and clients).
+
+## Implementation Details
+
+### Stage 1: Groth16 Checkpoint SNARKs
+
+1. Define a checkpoint interval (e.g., every 4 weeks or 1000 epochs).
+2. Develop a recursive Groth16 SNARK circuit that verifies multiple epoch SNARKs.
+3. Implement a process to generate and post checkpoint SNARKs to Celestia.
+4. Modify light clients to start syncing from the latest checkpoint SNARK.
+5. Ensure backwards compatibility for existing light clients (necessary rn?).
+
+### Stage 2: Transition to Advanced SNARK Systems
+
+1. Evaluate Nova and zkVM solutions (e.g., Nexus with HyperNova) for compatibility with our use case and future based rollup plans.
+2. Develop a prototype implementation using the chosen system.
+3. Create a migration strategy for existing proofs and clients.
+4. Implement additional proving capabilities required for based rollups (different ADR).
+5. Tests for the new system.
+
+## Why Nova (SuperNova / HyperNova) could be Beneficial for Us
+
+1. Efficient Aggregation: We aggregate many proofs into epoch SNARKs. Nova's folding scheme is particularly efficient for this use case, reducing our constraint complexity from possibly millions to tens of thousands.
+
+2. Scalability: As our system grows, the ability to efficiently aggregate proofs becomes crucial. Nova / SuperNova / HyperNova allows us to create "checkpoint SNARKs" more efficiently, enabling better long-term scalability.
+
+3. Reduced Resource Requirements: The dramatic reduction in constraint complexity translates to lower memory usage and faster proving times, allowing us to operate more efficiently and potentially on less powerful hardware.
+
+4. Flexibility: Nova's approach allows for more frequent aggregation and checkpoint creation, giving us more flexibility in how we structure our proof hierarchy.
+
+5. Future-proofing: Nova's design aligns well with potential future requirements, such as more complex proof structures or more frequent updates.
+
+## Alternatives Considered
+
+1. Using a different data availability layer with longer-term storage (@distractedm1nd told me already why not kyve?).
+2. Exploring other SNARK systems like Plonk or Halo2 (eventhough we do think Nova provides better efficiency for our specific use case of repeated proof aggregation).
+
+## Future Considerations
+
+While currently using Groth16, we should consider transitioning to more future-proof systems like Nexus zkVM with HyperNova for the following reasons:
```

Comment:
This is important but belongs in a different ADR

Reply:
To be honest, I thought this could be outlined here and then elaborated on in greater depth and detail below